jobs/buildhub/scrape_archives.py

```diff
@@ -180,16 +181,24 @@ def archive_url(product, version=None, platform=None, locale=None, nightly=None,
         raise ValueError("Could not fetch {}: {}".format(url, e))
 
 
-async def fetch_nightly_metadata(session, nightly_url):
+_nightly_metadata = {}
+
+async def fetch_nightly_metadata(session, product, nightly_url):
     """A JSON file containing build info is published along the nightly archive.
     """
-    # XXX: It is only available for en-US though. Should we use the same for every locale?
-    if "en-US" not in nightly_url:
+    if not is_release_filename(product, os.path.basename(nightly_url)):
         return None
```

Comment:
Why do we have to do this here? Can you explain in a comment? Why not do this in the calling code?